File: activities/07-parsing/tree/Java/TreeNum.java
//TreeNum:top//
//TreeNum:import//
import java.util.*;

// <tree>:TreeNum ::= <NUM>
public class TreeNum extends Tree /*TreeNum:class*/ {

    public static final String $className = "TreeNum";
    public static final String $ruleString =
        "<tree>:TreeNum ::= <NUM>";

    public Token num;

    public TreeNum(Token num) {
//TreeNum:init//
        this.num = num;
    }

    public static TreeNum parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<tree>:TreeNum", scn$.lno);
        Token num = scn$.match(Token.Match.NUM, trace$);
        return new TreeNum(num);
    }

//TreeNum//
}

File: activities/07-parsing/tree/Python/TreeNum.py
#TreeNum:top#
#TreeNum:import#

# <tree>:TreeNum ::= <NUM>
class TreeNum(Tree): #TreeNum:class#

    className = "TreeNum"
    ruleString = "<tree>:TreeNum ::= <NUM>"
    num = None

    def __init__(num):
        #TreeNum:init#
        self.num = num

#TreeNum#
